DS-970: Add Magic Nix Cache and other workflow changes

flexiondotorg opened the pull request:
An assortment of GitHub Workflow changes, potentially including:

- Enable DeterminateSystems/magic-nix-cache-action@main
- Reference all DeterminateSystems actions via @main
- Make update.yaml consistent across repos
- Remove unnecessary github-token: from nix-installer-action
- Update actions/checkout@v2 to actions/checkout@v3

flexiondotorg committed:
DS-970: Add Magic Nix Cache and other workflow changes

 An assortment of GitHub Workflow changes, potentially including:

- Enable DeterminateSystems/magic-nix-cache-action@main
- Reference all DeterminateSystems actions via @main
- Make update.yaml consistent across repos
- Remove unnecessary github-token: from nix-installer-action
- Update actions/checkout@v2 to actions/checkout@v3

Hoverbear commented:
The CI failed with an error that looks feasibly related to these changes, could you take a look? It might be solved with a `nix flake lock update`?

lheckemann committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'flake-utils':
    'github:numtide/flake-utils/c0e246b9b83f637f4681389ecabcb2681b4f3af0' (2022-08-07)
  → 'github:numtide/flake-utils/919d646de7be200f3bf08cb76ae1f09402b6f9b4' (2023-07-11)
• Added input 'flake-utils/systems':
    'github:nix-systems/default/da67096a3b9bf56a91d16901293e51ba5b49a27e' (2023-04-09)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/14ccaaedd95a488dd7ae142757884d8e125b3363' (2022-10-09)
  → 'path:/nix/store/pxdlbv3id7b3jmrzxpa5s6scc7xm50ax-source?lastModified=1688764204&narHash=sha256-FsvK+tIvelCI0tWwlMDKfiyb7P%2fKfxpGbXMrdCKiT8s=&rev=d8bb6c681cf86265fdcf3cc3119f757bbb085835' (2023-07-07)

lheckemann commented:
This is a bug in the flake checker, I've opened https://github.com/DeterminateSystems/flake-checker/issues/54

lucperkins committed:
Fix invalid flake.lock

lucperkins commented:
I'm not quite sure what the issue is here but I'm going to merge and address later as this is a mostly unused repo

lucperkins commented:
I'm not quite sure what the issue is here but I'm going to merge and address later as this is a mostly unused repo
